### DISCUSSION TITLE
Comparison of codeql and sonar qube for security

### DISCUSSION THREAD

**Discussion by dk98126** [community/community#23628] [New to GitHub]
<p>Hello guys!</p>
<p>I wonder if there are any papers or articles on codeql being compared to other tools like sonar, pvs studio or any other?</p>
<p>It would be really great if you could share link or name of such documents.</p>
<p>Thanks!</p>

**Comment by adityasharad** [comment, marked as answer]
<p>Hello. We do not publish public materials on comparisons with other tools.<br>
If your focus is public open-source repositories, I encourage you to try out code scanning with CodeQL as a way of evaluating it yourself: see <a href="https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning" class="inline-onebox">About code scanning - GitHub Docs</a> to get started.<br>
If you are interested in using CodeQL and GitHub Advanced Security for your organisation or private repositories, I encourage you to check out <a href="https://github.com/features/security" class="inline-onebox">Features · Security · GitHub</a>. That page includes links to this <a href="https://resources.github.com/whitepapers/Application-security-guide/">application security guide</a>, links to several other <a href="https://resources.github.com/whitepapers">whitepapers</a>, and contact information for the GitHub Advanced Security sales team, who can assist you with evaluating the technology on your organisation’s codebases.</p>

**Comment by mohammed-anwar-ul-haque-db** [comment]
Hello team,

Is there any comparison model between SonarQube and GitHub Checks API

**Comment by kimsyversen** [comment]
For those that came here hoping for answers, this may be insightful to you: https://www.researchgate.net/publication/372134053_Analysis_of_tools_for_static_security_testing_of_applications
